packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -1509,7 +1502,7 @@ class PubDependencyTree {
       // because they were omitted from pubspec.yaml used for 'pub upgrade' run.
       return result;
     }
-    for (final String dependency in _dependencyTree[package]) {
+    for (final String dependency in _dependencyTree[package]!) {
```

Comment:
Alternatively instead of the `containsKey`:
```dart
    final Set<String>? dependencies = _dependencyTree[package];
    if (dependencies == null) {
      // We have no transitive dependencies extracted for flutter_sdk packages
      // because they were omitted from pubspec.yaml used for 'pub upgrade' run.
      return result;
    }
    for (final String dependency in dependencies) {
```

Reply:
done